### ISSUE TITLE
Inclusion of "anchor" attribute in link is unnecessary in many examples

### ISSUE THREAD

**Issue by EskoDijk** [core-wg/resource-directory#281]
In many examples, the "anchor" attribute is included if the target URI of the link is an absolute URI. For example like below:

```
<coap://new.example.com:5684/sensors/temp>;ct=41;
    rt="temperature-c";if="sensor";
    anchor="coap://new.example.com:5684/",
```

Here the anchor is not needed because the origin of the target URI already specifies the context/anchor; per RFC 6690 https://tools.ietf.org/html/rfc6690#section-2.1 rule (b) .

For constrained devices and networks it is very important to minimize the payload sizes where possible, so I think the RD SHOULD in this case return the most efficient form possible by leaving out the anchor parameter per link where this is unnecessary to include.

New proposed text of above example:

```
<coap://new.example.com:5684/sensors/temp>;ct=41;
    rt="temperature-c";if="sensor",
```

And similar changes to be made for other examples.
Also I'd like to note that including the anchor additionally with the same information seems strange, given that the "hosts" relation type was defined as it is to get rid of the anchor in most cases - serving the needs of constrained systems.

**Comment by chrysn** [comment]
Result of the last interim's discussion was that yes we can go ahead with this (after checking through the implementations list again and making the adequate changes to the text).

I'm holding off the other "examples" fixes (#293, #286, #284, #283) for then to keep the diffs more readable.

**Comment by chrysn** [comment]
For my reference, what needs checking is the list at https://mailarchive.ietf.org/arch/msg/core/D_kuIq6QoBZ86FNniN7G_6d1Rtg against the current understanding.

**Comment by chrysn** [comment]
Fixed in eb62d317c4d9a650002bcee8c8fcc619134b7c4c
